core/embed/rust/src/ui/model_tr/component/mod.rs

```diff
@@ -1,21 +1,98 @@
 mod button;
-mod confirm;
-mod dialog;
-mod frame;
+mod button_controller;
+mod common;
+mod hold_to_confirm;
+mod input_methods;
 mod loader;
-mod page;
 mod result;
+mod welcome_screen;
+
+use super::theme;
+pub use button::{
+    Button, ButtonAction, ButtonActions, ButtonContent, ButtonDetails, ButtonLayout, ButtonMsg,
```

Comment:
Why is bootloader using `Button` but firmware `ButtonActions`+etc? Are you planning to use the latter in BL and making `Button` a private struct?

Reply:
Bootloader in this branch is not working properly, it still uses "raw" `Button` and `ButtonMsg`, even though `Button::event` does not return anything.

It is being fixed in https://github.com/trezor/trezor-firmware/pull/2959 (using `ButtonController`)

I will get rid of `ButtonMsg` here, so we will not forget about it